cpp/tests/test_refinement.cpp

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2022, Arthur Saliou, Anastasiia Kornilova
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <gtest/gtest.h>
+
+#include <deplex/config.h>
+#include <deplex/plane_extractor.h>
+#include <deplex/utils/depth_image.h>
+#include <deplex/utils/eigen_io.h>
+#include "../deplex/src/deplex/cell_segment_stat.h"
```

Comment:
If you want an ability to include private headers it's better to provide path to deplex sources via target_include_directories in test's CMakeLists